expression/aggregation/descriptor.go

```diff
@@ -43,6 +43,7 @@ type AggFuncDesc struct {
 }
 
 // NewAggFuncDesc creates an aggregation function signature descriptor.
+// this func cannot be called twice as the TypeInfer will change return type.
```

Comment:
Does this function change the return type for all the cases?

Reply:
[TypeInfer](https://github.com/pingcap/tidb/blob/155a6027db13e312e2b8d7d3b5a660a44e51f639/expression/aggregation/base_func.go#L87) change types especially for decimal type.

Reply:
The comment is a little confusing, I think by `change return type` you actually mean `change the input type`?

Reply:
> The comment is a little confusing, I think by `change return type` you actually mean `change the input type`?

done